lib/model/typing_status.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import '../api/model/events.dart';
+import 'narrow.dart';
+
+/// The view-model for tracking the typing status organized by narrows.
+class TypingStatus extends ChangeNotifier {
+  @visibleForTesting
+  final Map<SendableNarrow, Map<int, Timer>> typistIdsByNarrow = {};
+
+  final int selfUserId;
+  final Duration typingStartedExpiryPeriod;
+  final Duration typingStoppedWaitPeriod;
+  final Duration typingStartedWaitPeriod;
```

Comment:
I wonder if `typingStoppedWaitPeriod` and `typingStartedWaitPeriod` belong in this class, which is about reading and tracking users' typing status. I see that they're unused in this PR. These two values are about _sending_ the self-user's typing status, which seems maybe different enough that we'll end up using them in a separate subsystem in a different class.

Reply:
Hmm, good point. Let's leave them out then.